docs/sources/clients/k6/query-scenario.md

```diff
@@ -0,0 +1,234 @@
+---
+title: Query scenario
+---
+
+# Running the queries using xk6-loki extension
+
+## Test scenario
+
+To run the test, it's necessary to create JavaScript scenario. We have already created
+an [example](https://github.com/grafana/xk6-loki/blob/21307675b3e2a3e0a2b907d2131c9e906a6f6d60/examples/read-scenario.js), so you can reuse and extend it.
+
+This scenario allows to configure the ratio of each type of query and ratio of time ranges.
+
+## Ratio
+
+According to your needs, you can configure ratio of each type of query, ration of time ranges, etc.
+
+To do this you need:
+
+1. to define ratio config. The config it's just an array of items with fields:
+
+    - `ratio` - (number) `0 > ration < 1`
+    - `item` - (any)
+
+   Example of ratio config of query types:
+
+   ```javascript
+      const queryTypeRatioConfig = [
+        {
+          ratio: 0.1,
+          item: readLabels
+        },
+        {
+          ratio: 0.15,
+          item: readLabelValues
+        },
+        {
+          ratio: 0.05,
+          item: readSeries
+        },
+        {
+          ratio: 0.5,
+          item: readRange
+        },
+        {
+          ratio: 0.2,
+          item: readInstant
+        },
+      ];
+      ```
+
+   _Note: `item` here is just a function that will be executed if it is
+   selected. [See.](https://github.com/grafana/xk6-loki/blob/21307675b3e2a3e0a2b907d2131c9e906a6f6d60/examples/read-scenario.js#L139)_
+
+   According to this config, the ratio of requests during test run will be following:
+
+    * ~10% of labels requests
+    * ~15% of label values requests
+    * ~5% of requests for series
+    * ~50% of range queries
+    * ~20% of instant queries
+    
+2. to pass the config to the function `createSelectorByRatio` that will create a function that will consume random number and will choose `item` according to
+   defined ratio.
+
+   ```javascript
+   const createSelectorByRatio = (ratioConfig) => {
+     let ratioSum = 0;
+     const executorsIntervals = [];
+     for (let i = 0; i < ratioConfig.length; i++) {
+       executorsIntervals.push({
+         start: ratioSum,
+         end: ratioSum + ratioConfig[i].ratio,
+         item: ratioConfig[i].item,
+       })
+       ratioSum += ratioConfig[i].ratio
+     }
+     return (random) => {
+       if (random >= 1 || random < 0) {
+         fail(`random value must be within range [0-1)`)
+       }
+       const value = random * ratioSum;
+       for (let i = 0; i < executorsIntervals.length; i++) {
+         let currentInterval = executorsIntervals[i];
+         if (value < currentInterval.end && value >= currentInterval.start) {
+           return currentInterval.item
+         }
+       }
+     }
+   }
```

Comment:
I don't understand this and I doubt user will. Do we need this ? If yes we need to be able to explain more clearly what this does ?

Reply:
sure, I will put a comment on each line

Reply:
But also overall what is this function for ?